src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,190 @@
+    except process.CmdError as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
+
+def storage_format(hosts):
+    """ Execute format command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run format on.
+        path (str, optional): Path to dmg command binary. Defaults to "".
+
+    Returns:
+        False if issue running command. True otherwise.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+    dmg.request.value = "storage"
+    dmg.action.value = "format"
+    dmg.format.force.value = True
+
+    try:
+        result = dmg.run(sudo=True)
+    except process.CmdError as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
+
+def storage_prep(hosts, user=None, hugepages="4096", nvme=None,
+                 scm=None):
+    """ Execute prepare command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run prepare on.
+        path (str, optional): Path to dmg command binary. Defaults to "".
+
+    Returns:
+        False if issue running command. True otherwise.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+    dmg.request.value = "storage"
+    dmg.action.value = "prepare"
+    dmg.prepare.nvmeonly.value = nvme
+    dmg.prepare.scmonly.value = scm
+    dmg.prepare.targetuser.value = getpass.getuser() if user is None else user
+    dmg.prepare.hugepages.value = hugepages
+    dmg.prepare.force.value = True
+
+    try:
+        result = dmg.run()
+    except process.CmdError as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
+
+def storage_reset(hosts, user=None, hugepages="4096"):
+    """Execute prepare reset command through dmg tool to servers provided.
+
+    Args:
+        hosts ([type]): [description]
+        path (str, optional): [description]. Defaults to "".
+        user ([type], optional): [description]. Defaults to None.
+        hugepages (str, optional): [description]. Defaults to "4096".
+
+    Returns:
+        bool: False if issue running command. True otherwise.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+    dmg.request.value = "storage"
+    dmg.action.value = "prepare"
+    dmg.prepare.nvmeonly.value = True
+    dmg.prepare.targetuser.value = getpass.getuser() if user is None else user
+    dmg.prepare.hugepages.value = hugepages
+    dmg.prepare.reset.value = True
+    dmg.prepare.force.value = True
+
+    try:
+        result = dmg.run()
+    except process.CmdError as details:
```

Comment:
This particular exception is caught by `ExecutableCommand.run()` and in its place it will return a `CommandFailure` exception, which is what should be caught here.

Reply:
Will change